app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -235,8 +259,24 @@ class UserProfileFragment : Fragment() {
                 )
             }
         }
+    }
+
+    private fun signOut() {
+        AuthUI.getInstance()
+            .signOut(requireContext())
+            .addOnCompleteListener {
+                initSetup()
+                userType = UserType.LOGGED_OUT_SELF
+                setupViews()
+                setVisibilities()
+            }
+    }
```

Comment:
Since the signIn is in AuthHelper.kt, wouldn't it be cleaner to place this function there too?

Reply:
As when we click on signOut, the function touches the views, I couldn't put it into the AuthHelper. However, I can spend some time trying to figure out how I could do this.